lib/ansible/modules/remote_management/cisco/cisco_ucs_dns.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/python
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: cisco_ucs_dns
+short_description: Configures DNS on a Cisco UCS server.
+version_added: "2.4"
+description:
+   -  Configures DNS on a Cisco UCS server.
+options:
+    state:
+        description:
+         - if C(present), adds DNS server
+         - if C(absent), removes DNS server
+        choices: ['present', 'absent']
+        default: "present"
+    name:
+        description: ip address of dns server
+        required: true
+    descr:
+        description: description of server
+
```

Comment:
there are some missing docs e.g ucs_* params, these seems to be "generic" and it would make sense to add them to the doc fragments (for inclusion in other ucs modules) See https://github.com/ansible/ansible/tree/devel/lib/ansible/utils/module_docs_fragments